src/statemachine/RSocketStateMachine.cpp

```diff
@@ -774,12 +790,15 @@ void RSocketStateMachine::requestFireAndForget(Payload request) {
       streamsFactory().getNextStreamId(),
       FrameFlags::EMPTY,
       std::move(std::move(request)));
+  VLOG(3) << "Out: " << frame;
   outputFrameOrEnqueue(frameSerializer_->serializeOut(std::move(frame)));
 }
 
 void RSocketStateMachine::metadataPush(std::unique_ptr<folly::IOBuf> metadata) {
+  Frame_METADATA_PUSH metadataPushFrame(std::move(metadata));
+  VLOG(3) << "Out: " << metadataPushFrame;
   outputFrameOrEnqueue(
-      frameSerializer_->serializeOut(Frame_METADATA_PUSH(std::move(metadata))));
+      frameSerializer_->serializeOut(std::move(metadataPushFrame)));
```

Comment:
We need to keep it only if we call outputFrameOrEnqueue on frameTransport_ object, not on this object.
   frameTransport_->outputFrameOrEnqueue(frameSerializer_->serializeOut